content/en/blog/_posts/2023-04-04-in-place-pod-resize/index.md

```diff
@@ -0,0 +1,193 @@
+---
+
+**Author:** Vinay Kulkarni (Kubescaler Labs)
+
+Kubernetes v1.27 brings a new alpha feature that allows users to resize CPU
+and memory resources allocated to pods without restarting them. The `resources`
+field in a pod's containers are now mutable for `cpu` and `memory` resources.
+Additionally, a new `restartPolicy` for resize gives users control over how
+their containers are handled during resize.
+
+Kubernetes gets the actual CPU and memory requests and limits enforced on the
+running containers via a CRI (Container Runtime Interface) API call that queries
+the running container's status. The CRI API changes needed to support this
+feature were merged in v1.25 release.
+
+This also means that `resources` field in the pod spec can no longer be relied
+on as an indicator of the pod's actual resources. Monitoring tools and other
+such applications must look at new fields in the pod's status.
+
+
+## What's new in v1.27?
+
+Besides the aforementioned resize policy in the pod's spec, a new field named
+`allocatedResources` has been added to `containerStatuses` in the pod's status.
+This field reflects the node resources allocated to the pod's containers.
+
+In addition, a new field called `resources` has been added to the container's
+status. This field reflects the actual resource requests and limits configured
+on the running containers as reported by the container runtime.
+
+Lastly, a new field named `resize` has been added to the pod's status to show the
+status of the last requested resize. A value of `Proposed` is an acknowledgement
+of the requested resize and indicates that request was validated and recorded. A
+value of `InProgress` indicates that the node has accepted the resize request
+and is in the process of applying the resize request to the pod's containers.
+A value of `Deferred` means that the requested resize cannot be granted at this
+time, and the node will keep retrying. The resize may be granted when other pods
+leave and free up node resources. A value of `Infeasible` is a signal that the
+node cannot accommodate the requested resize. This can happen if the requested
+resize exceeds the maximum resources the node can ever allocate for a pod.
+
+
+## When to use this feature
+
+Here are a few examples where this feature may be useful:
+- Pod is running on node but with either too much or too little resources.
+- Pods are not being scheduled do to lack of sufficient CPU or memory in a
+cluster that is under-utilized.
+- Evicting certain pods that need more resources and scheduling them on
+bigger nodes is an expensive or disruptive operation when other pods on
+the nodes can be moved.
+
+
+## How to use this feature
+
+In order to use this feature in v1.27, the `InPlacePodVerticalScaling`
+feature gate must be enabled. A local cluster with this feature enabled
+can be started as shown below:
+
+```
+root@vbuild:~/go/src/k8s.io/kubernetes# FEATURE_GATES=InPlacePodVerticalScaling=true ./hack/local-up-cluster.sh
```

Comment:
If you've the appetite for it, document how to try it out with minikube, and ask to go later in the article schedule so that minikube is already released for v1.27 by the time you publish.

Reply:
Perhaps later, I have other priorities at this time.